plugins/rest/rest.go

```diff
@@ -39,6 +39,12 @@ type HTTPAuthPlugin interface {
 	Prepare(*http.Request) error
 }
 
+// HTTPHeaderAuthPlugin represents a mechanism to provide a header for HTTP authentication for a REST service
+type HTTPHeaderAuthPlugin interface {
```

Comment:
Can you move the interface to the `download` package instead and make it private?
I think the `download` package would be the better place to define it since it's the only place where it's used. It also allows us to keep it private and not further extend the public API.

Reply:
done